mesonbuild/coredata.py

```diff
@@ -1016,15 +1016,24 @@ def set_default_options(self, default_options: T.MutableMapping[OptionKey, str],
 
         self.set_options(options, subproject=subproject, first_invocation=env.first_invocation)
 
-    def add_compiler_options(self, options: 'MutableKeyedOptionDictType', lang: str, for_machine: MachineChoice,
-                             env: 'Environment') -> None:
+    def add_compiler_options(self, options: MutableKeyedOptionDictType, lang: str, for_machine: MachineChoice,
+                             env: Environment, subproject: str) -> None:
         for k, o in options.items():
             value = env.options.get(k)
             if value is not None:
                 o.set_value(value)
-                self.options[k] = o  # override compiler option on reconfigure
+                if not subproject:
+                    self.options[k] = o  # override compiler option on reconfigure
             self.options.setdefault(k, o)
```

Comment:
This set's the subproject='' key to the subproject value, that that doesn't seem right.